tools/link.py

```diff
@@ -606,25 +606,88 @@ def include_and_export(name):
       settings.EXPORTED_RUNTIME_METHODS += ['ExitStatus']
 
 
+def set_initial_memory():
+  # INITIAL_HEAP cannot be used when the memory object is created in JS.
+  if settings.IMPORTED_MEMORY:
+    if 'INITIAL_HEAP' not in user_settings:
```

Comment:
Flip this condition so that its a positive test and then `exit_with_error` will early return meaning you don't need the `else` anymore.